src/pluggy/_manager.py

```diff
@@ -181,7 +192,19 @@ def parse_hookimpl_opts(self, plugin: _Plugin, name: str) -> HookimplOpts | None
         customize how hook implementation are picked up. By default, returns the
         options for items decorated with :class:`HookimplMarker`.
         """
-        method: object = getattr(plugin, name)
+
+        if _attr_is_property(plugin, name):
+            # @property methods can have side effects, and are never hookimpls
+            return None
+
+        method: object
+        try:
+            method = getattr(plugin, name)
+        except AttributeError:
+            # AttributeError: '__signature__' attribute of 'plugin' is class-only
```

Comment:
If I understand correctly, this is to handle a proxy object which returns some attribute from `dir(obj)` but `getattr(obj, attr)` raises `AttributeError`.

To me this seems like a malfunctioning proxy implementation, which should be fixed by properly overriding [`__dir__`](https://docs.python.org/3/library/functions.html#dir) or similar.

Reply:
That's correct @bluetech, but unfortunatley some common classes/module namespaces in the wild don't behave stricly like a barebones class or `MappingType`s.  e.g. pydantic models, benedict dicts, django model classes.

The case is easy to catch though, becauase getting an `AttributeError` at that point while iterating through __dir__.keys() is surprising/rare:

```pythno
try:
    method = getattr(plugin, name)
except AttributeError:   # means it's not a normal object
    # oops, we actually want the base class it's wrapping,
    # not the wrapper proxy object.
    method = getattr(type(plugin), name)
```

With the patch you can define classes with `@hookimpl` methods on them like normal:
```python
from django.db import models
from pydantic import BaseModel

from .plugin_spec import hookimpl

class FancyGizmoPlugin(BaseModel): 
    # or any other "wrapped" ^ base class from django, pydantic, dataclass, etc.
    # numpy 
    
	name: 'Some Fancy New Gizmo!'
	
	@hookimpl
	def get_all_plugins(self):
	    return {self.name: self}
	    
    @hookimpl
    def finished_loading(self):
		print(f'{self.name} finished loading succesfully!')
	
    @hookimpl
    def do_some_cool_thing(self):
        ...
```

Reply:
> Unfortunately you cant "fix" __dir__ on these classes becauase they are managed by the pydantic/django/etc. and it would break the library.

I think it's a bug in pydantic (returning an attr from `__dir__` but not being able to get it), and if you report it they might fix it. But as much as I dislike workarounds, I guess we can live with one for this.

Do you have a use case for the `method = getattr(type(plugin), name)` fallback? It doesn't make sense to me to grab from the class. Can you live with the fix being to change `getattr(plugin, name)` to `getattr(plugin, name, None)`?

Reply:
The only case it fixes is if an instance proxy sets up some dynamic field at `__init__` time that shadowed a hookimpl method defined on the base class. It would prevent pluggy from detecting the hookimpl on the class, but I could see that being intentional/not worth edge-casing, so I can live without it if you'd prefer to remove it.

Reply:
Let's start with `getattr(plugin, name, None)` and consider something more elaborate separately if someone needs it.

Reply:
Ok PR is updated, changed to return `None` for now instead of falling back to `getattr(type(plugin), name).